ddsketch/store/store_test.go

```diff
@@ -777,6 +777,40 @@ func TestSerialization(t *testing.T) {
 	}
 }
 
+func TestSparseStoreSerialization(t *testing.T) {
+	nTests := 100
+	// Store indices are limited to the int32 range
+	var values []int32
+	f := fuzz.New().NilChance(0).NumElements(10, 1000)
+	for i := 0; i < nTests; i++ {
+		f.Fuzz(&values)
+		store := NewSparseStore()
+		for _, v := range values {
+			store.Add(int(v))
+		}
+		deserializedStore := SparseStoreFromProto(store.ToProto())
+		assert.Equal(t, store, deserializedStore)
+	}
+}
+
+func TestBufferPaginatedStoreSerialization(t *testing.T) {
+	nTests := 100
+	// Store indices are limited to the int32 range
+	var values []int32
+	f := fuzz.New().NilChance(0).NumElements(10, 1000)
+	for i := 0; i < nTests; i++ {
+		f.Fuzz(&values)
+		store := NewBufferedPaginatedStore()
+		for _, v := range values {
+			store.Add(int(v))
+		}
+		deserializedStore := BufferPaginatedStoreFromProto(store.ToProto())
+		sort.Ints(store.buffer)
+		sort.Ints(deserializedStore.buffer)
+		assert.Equal(t, store, deserializedStore)
+	}
+}
```

Comment:
This test is flaky. Working on figuring out why, but @CharlesMasson I may need your help since you know the details of the `BufferPaginatedStore` the best.